// ink-big-text/dist.js
"use strict";

var _react = _interopRequireDefault(require("react"));

var _ink = require("ink");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _cfonts = _interopRequireDefault(require("cfonts"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const BigText = ({
  text,
  ...props
}) => /*#__PURE__*/_react.default.createElement(_ink.Text, null, _cfonts.default.render(text, props).string);

BigText.defaultProps = {
  font: 'block',
  align: 'left',
  colors: ['system'],
  backgroundColor: 'transparent',
  letterSpacing: 1,
  lineHeight: 1,
  space: true,
  maxLength: 0
};
BigText.propTypes = {
  text: _propTypes.default.string.isRequired,
  font: _propTypes.default.oneOf(['block', 'slick', 'tiny', 'grid', 'pallet', 'shade', 'simple', 'simpleBlock', '3d', 'simple3d', 'chrome', 'huge']),
  align: _propTypes.default.oneOf(['left', 'center', 'right']),
  colors: _propTypes.default.arrayOf(_propTypes.default.string),
  backgroundColor: _propTypes.default.oneOf(['transparent', 'black', 'red', 'green', 'yellow', 'blue', 'magenta', 'cyan', 'white']),
  letterSpacing: _propTypes.default.number,
  lineHeight: _propTypes.default.number,
  space: _propTypes.default.bool,
  maxLength: _propTypes.default.number
};
module.exports = BigText;
